pkg/agent/openflow/client.go

```diff
@@ -104,10 +103,14 @@ type Client interface {
 	// otherwise the installation will fail.
 	// nodeLocalExternal represents if the externalTrafficPolicy is Local or not. This field is meaningful only when
 	// the svcType is NodePort or LoadBalancer.
-	InstallServiceFlows(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, nodeLocalExternal bool, svcType v1.ServiceType) error
+	// nestedEndpointSupport represents if the Service has the Endpoints which is other Service's ClusterIP.
+	InstallServiceFlows(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol, affinityTimeout uint16, nodeLocalExternal bool, svcType v1.ServiceType, nestedEndpointSupport bool) error
 	// UninstallServiceFlows removes flows installed by InstallServiceFlows.
 	UninstallServiceFlows(svcIP net.IP, svcPort uint16, protocol binding.Protocol) error
 
+	// InstallServiceClusterIPFlows install flows for accessing Endpoints which is Service's ClusterIP, and has an action
+	// to Service's corresponding group.
+	InstallServiceClusterIPFlows(svcIP net.IP, svcPort uint16, protocol binding.Protocol, groupID binding.GroupIDType) error
```

Comment:
Thought it over again, perhaps it's simpler to just generate one more flow in `InstallServiceFlows ` directly.

```
func InstallServiceFlows() {
...
if svcType == v1.ServiceTypeClusterIP && !nested {
      flows = append(flows, c.featureService. endpointRedirectFlowForServiceIP(svcIP, svcPort, protocol, groupID))
}
...
}

```

Reply:
Make sense, I will update it.